internal/cmd/xmrswap/main.go

```diff
@@ -0,0 +1,1221 @@
+//go:build libsecp256k1
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/config"
+	dcradaptor "decred.org/dcrdex/internal/adaptorsigs/dcr"
+	"decred.org/dcrdex/internal/libsecp256k1"
+	"decred.org/dcrwallet/v4/rpc/client/dcrwallet"
+	dcrwalletjson "decred.org/dcrwallet/v4/rpc/jsonrpc/types"
+	"github.com/agl/ed25519/edwards25519"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/chaincfg/v3"
+	"github.com/decred/dcrd/dcrec"
+	"github.com/decred/dcrd/dcrec/edwards/v2"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/decred/dcrd/rpcclient/v8"
+	"github.com/decred/dcrd/txscript/v4"
+	"github.com/decred/dcrd/txscript/v4/sign"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/slog"
+	"github.com/dev-warrior777/go-monero/rpc"
+	"github.com/fatih/color"
+	"github.com/haven-protocol-org/monero-go-utils/base58"
+)
+
+// TODO: Verification at all stages has not been implemented yet.
+
+// fieldIntSize is the size of a field element encoded
+// as bytes.
+const (
+	fieldIntSize = 32
+	dcrAmt       = 7_000_000 // atoms
+	xmrAmt       = 1_000     // 1e12 units
+	dumbFee      = int64(6000)
+)
+
+var (
+	homeDir    = os.Getenv("HOME")
+	dextestDir = filepath.Join(homeDir, "dextest")
+	bobDir     = filepath.Join(dextestDir, "xmr", "wallets", "bob")
+	curve      = edwards.Edwards()
+)
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	if err := run(ctx); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+	}
+}
+
+type walletClient = dcrwallet.Client
+
+type combinedClient struct {
+	*rpcclient.Client
+	*walletClient
+	chainParams *chaincfg.Params
+}
+
+func newCombinedClient(nodeRPCClient *rpcclient.Client, chainParams *chaincfg.Params) *combinedClient {
+	return &combinedClient{
+		nodeRPCClient,
+		dcrwallet.NewClient(dcrwallet.RawRequestCaller(nodeRPCClient), chainParams),
+		chainParams,
+	}
+}
+
+type client struct {
+	xmr *rpc.Client
+	dcr *combinedClient
+
+	kbvf, kbsf, kbvl, kbsl, vkbv *edwards.PrivateKey
+	pkbsf, pkbs                  *edwards.PublicKey
+	kaf, kal                     *secp256k1.PrivateKey
+	pkal, pkaf, pkasl, pkbsl     *secp256k1.PublicKey
+	kbsfDleag, kbslDleag         [libsecp256k1.ProofLen]byte
```

Comment:
It's really hard to follow the code with these variable names. Something like `ourSpendKey`, `counterPartySpendKey`, etc. would make it way more understandable.

Reply:
Better clean these up on top of #2936? No reason to create conflicts there imo.